src/audio/buffers/comp_buffer.c

```diff
@@ -170,7 +171,7 @@ static const struct source_ops comp_buffer_source_ops = {
 	.set_alignment_constants = audio_buffer_source_set_alignment_constants,
 };
 
-static const struct sink_ops comp_buffer_sink_ops = {
+APP_TASK_DATA static const struct sink_ops comp_buffer_sink_ops = {
```

Comment:
This seems a bit amazing that APP_TASK_DATA is only needed for these two structs. Why these two?

Reply:
These two structs contain function pointers used by the sink/source API to operate on audio buffers. The DP thread in userspace uses this API to process audio, so it needs access to these structs to call the correct functions. That’s why they must reside in a memory partition (`common_partition`) accessible to the DP thread - hence the use of APP_TASK_DATA.

There may be more structs like this in the future, but we're migrating them incrementally to avoid bloating the PR and to keep changes reviewable.